File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/Group.java
package ru.nikshlykov.englishwordsapp.db;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "Groups")
public class Group {

    @PrimaryKey()
    @NonNull
    @ColumnInfo(name = "_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "GroupName")
    public String name;

    public static class GroupsTable {
        // Названия таблицы групп и её колонок
        static final String TABLE_GROUPS = "Groups";
        static final String TABLE_GROUPS_COLUMN_ID = "_id";
        static final String TABLE_GROUPS_COLUMN_GROUPNAME = "GroupName";
    }
}

File: core-db/src/main/java/ru/nikshlykov/data/database/models/Group.kt
package ru.nikshlykov.data.database.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "Groups")
data class Group(
  @PrimaryKey @ColumnInfo(name = "_id") var id: Long = 0,
  @ColumnInfo(name = "GroupName") var name: String = ""
) {
  companion object {
    const val GROUP_FOR_NEW_SUBGROUPS_ID = -1L
  }
}